tests/cli/test_traces.py

```diff
@@ -0,0 +1,222 @@
+        # Check that either the table output or the values appear in output
+        assert "tr-123" in result.output
+        assert "OK" in result.output
+
+
+def test_get_command_with_fields(runner):
+    """Test get command with field selection."""
+    mock_trace = {
+        "info": {"trace_id": "tr-123", "state": "OK"},
+        "data": {"spans": [{"name": "test_span"}]},
+    }
+
+    # Create a mock trace object with a to_dict method
+    mock_trace_obj = mock.Mock()
+    mock_trace_obj.to_dict.return_value = mock_trace
+
+    # Patch the TracingClient at the module level where it's imported
+    with mock.patch("mlflow.cli.traces.TracingClient") as mock_client:
+        mock_instance = mock.Mock()
+        mock_client.return_value = mock_instance
+        mock_instance.get_trace.return_value = mock_trace_obj
+
+        result = runner.invoke(
+            commands,
+            ["get", "--trace-id", "tr-123", "--extract-fields", "info.trace_id"],
+        )
+
+        assert result.exit_code == 0
+
+        # Parse JSON output
+        output_json = json.loads(result.output)
+        assert output_json == {"info": {"trace_id": "tr-123"}}
+
+
+def test_delete_command(runner):
+    """Test delete command."""
+    # Patch the TracingClient at the module level where it's imported
+    with mock.patch("mlflow.cli.traces.TracingClient") as mock_client:
+        mock_instance = mock.Mock()
+        mock_client.return_value = mock_instance
+        mock_instance.delete_traces.return_value = 5
+
+        result = runner.invoke(
+            commands,
+            ["delete", "--experiment-id", "1", "--trace-ids", "tr-1,tr-2,tr-3"],
+        )
+
+        assert result.exit_code == 0
+        assert "Deleted 5 trace(s)" in result.output
+
+
+def test_field_validation_error(runner):
+    """Test that invalid fields produce helpful error messages."""
+    mock_trace = {"info": {"trace_id": "tr-123"}}
+
+    # Create a mock trace object with a to_dict method
+    mock_trace_obj = mock.Mock()
+    mock_trace_obj.to_dict.return_value = mock_trace
+
+    mock_result = PagedList([mock_trace_obj], None)
+
+    # Patch the search_traces method directly
+    with mock.patch("mlflow.cli.traces.TracingClient.search_traces", return_value=mock_result):
+        result = runner.invoke(
+            commands,
+            ["search", "--experiment-id", "1", "--extract-fields", "invalid.field"],
+        )
+
+        assert result.exit_code != 0
+        assert "Invalid field path" in result.output
+        # Check for the tip about verbose mode
+        assert "--verbose" in result.output
+
+
+def test_field_validation_error_verbose_mode(runner):
+    """Test that verbose mode shows all available fields."""
+    mock_trace = {
+        "info": {
+            "trace_id": "tr-123",
+            "state": "OK",
+            "request_time": 1700000000000,
+            "assessments": [{"id": "a1"}],
+        },
+        "data": {"spans": [{"name": "span1"}]},
+    }
```

Comment:
```suggestion
    mock_trace = Trace.from_dict({
        "info": {
            "trace_id": "tr-123",
            "state": "OK",
            "request_time": 1700000000000,
            "assessments": [{"id": "a1"}],
        },
        "data": {"spans": [{"name": "span1"}]},
    })
```

Do we need to use mock? can we use a real trace object?